### PR TITLE
Shopping cart ui improvements

### PR DESCRIPTION
**Pull request by tadeuszwojcik**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**

**Other information:**
closes https://github.com/deity-io/falcon/issues/136

### CODE REVIEW

File: packages/falcon-ecommerce-uikit/src/Cart/CartMutation.tsx [hunk 1]
```diff
@@ -17,20 +18,43 @@ export const ADD_TO_CART = gql`
...
+        data: {
+          removeCartItem: { itemId }
+        }
+      }: any
+    ) => {
+      const data = store.readQuery({
+        query: GET_CART
+      });
+
+      data.cart.items = data.cart.items.filter((item: any) => item.itemId !== itemId);
```
Comment (Borales) [inline packages/falcon-ecommerce-uikit/src/Cart/CartMutation.tsx]: @tadeuszwojcik `refetchQueries: ['Cart']` - doesn't it fetch the cart data from the remote backend already? or is it done for "optimistic" response?

Comment (tadeuszwojcik) [reply packages/falcon-ecommerce-uikit/src/Cart/CartMutation.tsx]: There are two queries, `Cart` and `Minicart`. It used to be that delete wasn't `optimistic` so user had to wait until server responds with refetch for `Minicart` now it's handled optimistically and deleting from mini cart is instant for end user...but we still need to refetch `Cart` query (for full Cart page) as it has things like total price, discounts etc which are calculated on the backend.

File: packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx [hunk 2]
```diff
@@ -47,6 +47,31 @@ const miniCartProductTheme: DefaultThemeProps = {
...
+const miniCartLayout: DefaultThemeProps = {
+  miniCartLayout: {
+    display: 'grid',
+    gridRowGap: 'md',
+
+    // prettier-ignore
+    gridTemplate: toGridTemplate([
+      ['1fr',                    '30px'                         ],
+      [MiniCartLayoutArea.title, MiniCartLayoutArea.close,      ],
+      [MiniCartLayoutArea.items, MiniCartLayoutArea.items, '1fr'],
```
Comment (mateuszwozniak) [inline packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx]: What does that additional `'1fr'` at the end of this row?

Comment (tadeuszwojcik) [reply packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx]: It's row height, by default each row height is set to 'auto' implicitly, but here we want items row to be as high as possible (so checkout button is always at the bottom and title at the top).
There's also additional code that makes items row scrollable so even if there are many items in minicart checkout and title are always visible and only items area scrolls, does it make sense?

Comment (mateuszwozniak) [reply packages/falcon-ecommerce-uikit/src/MiniCart/MiniCart.tsx]: Yes, clear. Thanks for the explanation :)

**Pull request merged**